File: LOO 2°Parcial/Tarea 4/TrabajadorComision.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package empleadospolimorfismo;

/**
 * 
 * @author Dimitri Medina Novelo
 */
public class TrabajadorComision extends Empleado {
    protected double salarioBase, porcentajeComision, montoVentas;

        public TrabajadorComision()
        {
            super();
            this.salarioBase = 0;
            this.porcentajeComision = 0;
            this.montoVentas = 0;
        }

        public TrabajadorComision(String nombre, double salario, double comision, double ventas) 
        {
            super(nombre);
            this.salarioBase = salario > 0 ? salario : 0;
            this.porcentajeComision = comision > 0 && comision <=1 ? comision : 0;
            this.montoVentas = ventas > 0 ? ventas : 0;
        }
        
        public double getSalarioBase()
        {
            return this.salarioBase;
        }
        
        public void setSalarioBase(double salario)
        {
            this.salarioBase = salario > 0 ? salario : 0;
        }

        public double getPorcentajeComision()
        {
            return this.porcentajeComision;
        }
        
        public void setPorcentajeComision(double comision)
        {
            this.porcentajeComision = comision > 0 && comision <= 1 ? comision : 0;
        }

        public double getMontoVentas()
        {
            return this.montoVentas;
        }
        
        public void setMontoVentas(double ventas)
        {
            this.montoVentas = ventas > 0 ? ventas : 0;
        }
        
        @Override
        public double pagoDiario()
        {
            return this.salarioBase + this.porcentajeComision + this.montoVentas;
        }
        
        @Override 
        public void imprimir()
        {
            System.out.println("Empleado tipo TRABAJADOR POR COMISION");
            System.out.println("NOMBRE: " + this.nombre + "\nSALARIO: " + this.salarioBase + "\nPORCENTAJE DE COMISION: " + this.porcentajeComision*100 + "\nMONTO DE VENTAS: " + this.montoVentas + "\nPAGO DIARIO: " + this.pagoDiario());
        }
}

File: LOO 2°Parcial/Tarea 4/TrabajadorComision.py
# -*- coding: utf-8 -*-
"""
Created on Fri Mar 19 12:23:11 2021

@author: MedNo
"""

import Empleado as Emp

class TrabajadorComision(Emp.Empleado):
    def __init__(self, nombre="", salario=0, comision=0, ventas=0):
        Emp.Empleado.__init__(self,nombre)
        if salario>0:
            self.__salariobase = salario
        else:
            self.__salariobase = 0
        if comision>=0 and comision<=1:
            self.__porcentajecomision = comision
        else:
            self.__porcentajecomision = 0
        if ventas>0:
            self.__montoventas = ventas
        else:
            self.__montoventas = 0
            
    @property 
    def salariobase(self):
        return self.__salariobase
    
    @salariobase.setter
    def salariobase(self, salario):
        if salario>0:
            self.__salariobase = salario
        else:
            self.__salariobase = 0
    
    @property
    def porcentajecomision(self):
        return self.__porcentajecomision
    @porcentajecomision.setter
    def porcentajecomision(self, comision):
        if comision>=0 and comision<=1:
            self.__porcentajecomision = comision
        else:
            self.__porcentajecomision = 0
    
    @property 
    def montoventas(self):
        return self.__montoventas
    
    @montoventas.setter
    def montoventas(self, ventas):
        if ventas>0:
            self.__montoventas = ventas
        else:
            self.__montoventas = 0
            
    def pagoDiario(self):
        return self.__salariobase + self.__porcentajecomision * self.__montoventas
    
    def imprimir(self):
        print("Empleado tipo TRABAJADOR POR COMISIÓN");
        print("NOMBRE: " + self.nombre + "\nSALARIO BASE: " + str(self.__salariobase) + "\nPORCENTAJE COMISIÓN: " + str(self.__porcentajecomision*100) + "\nMONTO VENTAS: " + str(self.__montoventas) + "\nPAGO DIARIO: " + str(self.pagoDiario()))
        